jena-fuseki1/src/test/java/org/apache/jena/fuseki/ServerCtl.java

```diff
@@ -18,39 +18,113 @@
 
 package org.apache.jena.fuseki;
 
+import static org.apache.jena.fuseki.ServerCtl.ServerScope.CLASS ;
+import static org.apache.jena.fuseki.ServerCtl.ServerScope.SUITE ;
+import static org.apache.jena.fuseki.ServerCtl.ServerScope.TEST ;
+
+import java.io.IOException ;
+import java.net.ServerSocket ;
+import java.util.concurrent.atomic.AtomicInteger ;
+
 import org.apache.http.client.HttpClient ;
 import org.apache.http.impl.client.CloseableHttpClient ;
 import org.apache.jena.atlas.io.IO ;
+import org.apache.jena.fuseki.server.DatasetRegistry ;
 import org.apache.jena.riot.web.HttpOp ;
+import org.apache.jena.sparql.core.DatasetGraph ;
+import org.apache.jena.sparql.core.DatasetGraphFactory ;
+import org.apache.jena.sparql.modify.request.Target ;
+import org.apache.jena.sparql.modify.request.UpdateDrop ;
+import org.apache.jena.system.Txn ;
+import org.apache.jena.update.Update ;
+import org.apache.jena.update.UpdateExecutionFactory ;
+import org.apache.jena.update.UpdateProcessor ;
 
+/**
+ * Manage a single server for use wth tests. It supports three modes:
+ * <ul>
+ * <li>One server for a whole test suite
+ * <li>One server per test class
+ * <li>One server per individual test
+ * </ul>
+ * One server per individual test can be troublesome due to connections not closing down
+ * fast enough (left in TCP state {@code TIME_WAIT} which is 2 minutes) and also can be slow.
+ * One server per test class is a good compromise. 
+ * <p> The data in the server is always reseet between tests.
+ * <p>
+ * Using a connection pooling HttpClient (see {@link HttpOp#createPoolingHttpClient()}) is important,
+ * both for test performance and for reducing the TCP connection load on the operarting system.  
+ * <p>
+ * Usage:
+ * </p>
+ * <p>
+ * In the test suite, put:
+ * 
+ * <pre>
+ *  {@literal @BeforeClass} static public void beforeSuiteClass() { ServerCtl.ctlBeforeTestSuite(); } 
+ *  {@literal @AfterClass}  static public void afterSuiteClass()  { ServerCtl.ctlAfterTestSuite(); }
+ * </pre>
+ * <p>
+ * In the test class, put:
+ * <pre>
+ * {@literal @BeforeClass} public static void ctlBeforeClass() { ServerCtl.ctlBeforeClass(); }
```

Comment:
I doubt JUnit would cope with that - super classes yes, mixins no.

1. JUnit4 depends on Java 1.5 (ceratinly not Java8) so it probably does not even check interfaces.
2. Even if it did, there are now ordering issues. Multiple _BeforeClass_ happen in super-class order. If they can be inherited via multiple routes ... it gets interesting.

More generally, the test have the effect of those calls - and may have to achieve to work with their own _BeforeClass_/_AfterClass_.